src/uwtools/config.py

```diff
@@ -51,115 +54,84 @@
+    A base class specifying (and partially implementing) methods to read, manimulate, and write
+    several configuration-file formats.
     """
 
-    def __init__(self, config_path: Optional[str] = None, log_name: Optional[str] = None) -> None:
+    def __init__(self, config_path: str, log_name: Optional[str] = None) -> None:
         """
-        Parameters
-        ----------
-        config_path
-            See class-level docstring
-        log_name
-            Name to be used in log messages
+        Construct a Config object.
+
+        :param config_path: Path to the config file to load.
+        :param log_name: Name of logger object to log to.
         """
         super().__init__()
         self.config_path = config_path
         self.log = logging.getLogger(log_name)
+        self.update(self._load(self.config_path))
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         """
-        This method will return configure contents.
+        The string representation of a Config object.
         """
         return json.dumps(self.data)
 
     # Private methods
 
     @abstractmethod
-    def _load(self, config_path=None):
+    def _load(self, config_path: str) -> dict:
         """
-        Interface to load a config file given the config_path attribute, or optional config_path
-        argument.
+        Reads and parses a config file.
 
-        Returns a dict object.
-        """
+        Returns the result of loading and parsing the specified config file.
 
-    def _load_paths(self, filepaths):
+        :param config_path: Path to config file to load.
         """
-        Given a list of filepaths, load each file in the list and return a dictionary that includes
-        the parsed contents of the collection of files.
+
+    def _load_paths(self, filepaths: List[str]) -> dict:
         """
+        Merge and return the contents of a collection of config files.
 
+        :param filepaths: Paths to the config files to read and merge.
+        """
         cfg = {}
         for filepath in filepaths:
             if not os.path.isabs(filepath):
-                assert self.config_path is not None
                 filepath = os.path.join(os.path.dirname(self.config_path), filepath)
             cfg.update(self._load(config_path=filepath))
         return cfg
 
     # Public methods
 
-    def compare_config(self, user_dict, base_dict=None):
+    def compare_config(self, dict1: dict, dict2: Optional[dict] = None) -> None:
         """
-        Assuming a section, key/value structure of configuration types, compare the dictionary to
-        the values stored in the external file.
-        """
-        if base_dict is None:
-            base_dict = self.data
+        Compare two config dictionaries.
+
+        Assumes a section/key/value structure.
 
+        :param dict1: The first dictionary.
+        :param dict2: The second dictionary.
+        """
```

Comment:
I think that may have been derived from phrasing in the source ticket, something likely "compare a user-provided dict to a base reference dict". Since it doesn't exactly specify which is "true" in the logic, this is probably as good as anything

Reply:
That makes sense, thanks. If we were printing a report that referred to the dictionaries with special names (like `base` or `known good` or something) then I think it'd make sense to mirror that terminology in the variable names. But given how generic things are, I think I'll keep the generic names if no one objects.